gui/options.cpp

```diff
@@ -2116,7 +2116,9 @@ void GlobalOptionsDialog::build() {
 	Common::String iconPath(ConfMan.get("iconspath", _domain));
 	Common::String extraPath(ConfMan.get("extrapath", _domain));
 
-	if (savePath.empty() || !ConfMan.hasKey("savepath", _domain)) {
+	if (ConfMan.hasKey("savepath", Common::ConfigManager::kSessionDomain)) {
```

Comment:
I do not like this hardcoding. Instead of making it specific for the savepath, I would rather make it generic, for _all_ keys which are specified by the command line. Examples are `-c`, `-p`, `-g`, `-f` etc.

E.g. you may expand `ConfMan.hasKey()` but then add additional signage that those keys come from the command line. The current approach will _modify_ the path value in the GUI, and effectively prevent path having " (temporary)" string in many languages. 

Rather than modifying the string, you could consider, say, modifying its color and adding an explanation to the tooltip.

Reply:
I didn't think about the fact that now a valid path that contains " (temporary)" could not be saved. I will try to implement it with color.  
The other part I don't think I fully understand. Are you asking for a new interface in the config manager to check if a specific command line key has been specified (like maybe an overload of `hasKey` that takes and enum)?